### DISCUSSION TITLE
Aiutante come nome di file

### DISCUSSION THREAD

**Discussion by teeodor** [indomus/forum#2265] [HUB: Home Assistant]
Ciao a tutti, ho un problema che non riesco ad arrivarci in fondo.
Praticamente nella cartella www ho 6 file numerati da 1 a 6 con estensione jpg da usare come sfondo.
Ho creato un sensore random che mi genera un numero da 1 a 6 , poi ho creato 2 aiutanti, uno numerico e uno di testo che ogni giorno pescano il numero random.
Io non riesco a trovare la stringa corretta per mettere una di queste variabili come nome del file.
Questa è quello che ho trovato in questi giorni navigando, ma non funziona 
image: /local/{{states('input_text.prova')}}.jpg
ho provato anche con image: /local/{{states('sensor.random_sensor')}}.jpg
Non so più come fare

**Comment by snakuzzo** [comment]
ma dove vorresti mettere questo template? 
Chiedo perché non è che puoi usare i template a piacimento e ovunque...devi vedere se per quello che stai usando è previsto l'utilizzo.

Detto ciò, nel caso in cui fosse possibile utilizzare il template...controlla che il nome del file non abbia già anche l'estensione `.jpg`

```
image: "/local/{{ states('input_text.prova') }}.jpg"
```

...aiutati con i Developer Tools

**Reply by teeodor** [reply]
Lo dovrei usare in una immagine.
I file si chiamano numero+estensione, per esempio 1.jpg
Ovviamente se metto local/1.jpg funziona

**Reply by snakuzzo** [reply]
Hai verificato con i dev tools come ti ho detto?
Fai vedere come hai definito l'entità

**Comment by teeodor** [comment]
I devo tools non li ho mai usati.
L'ho impostato su configuration.yaml
sensor:
  - platform: random
    minimum: 1
    maximum: 6

**Reply by teeodor** [reply]
Forse ho capito cosa intendi per i developer tools
Inserisco questo
image: /local/{{ states('input_text.prova') }}.jpg
Il risultato è questo
image: /local/5.jpg

**Reply by teeodor** [reply]
Però il tipo di risultato è "string"

**Reply by snakuzzo** [reply]
Io non parlavo del sensore, ma voglio capire "DOVE" vuoi usare quel template.

**Comment by teeodor** [comment]
Lo devo usare in un elementi di immagine che non sto qua a copiare perché avrà 2 righe.
Sto facendo le prove nel modo più semplice con una semplice immagine
type: picture
image: /local/{{ states('input_text.prova') }}.jpg
tap_action:
  action: none
hold_action:
  action: none

**Reply by snakuzzo** [reply]
Mi sa che in quel tipo di card non puoi usare un template per valorizzare `image: `

**Reply by teeodor** [reply]
Sicuro?
Io l'ho provato in mille card e non va in nessuna.
In più ho visto gente che ci è riuscita

**Reply by snakuzzo** [reply]
devi leggere la doc delle card. Se si possono usare i template, solitamente c'è scritto
Se la gente ci è riuscita, vedi come ha fatto

**Reply by snakuzzo** [reply]
e vedi bene se usavano custom card

**Reply by teeodor** [reply]
Ho visto e infatti non capisco perché non mi va.
Nessuna custom card

**Reply by snakuzzo** [reply]
mi fai vedere cosa hai visto ?

**Reply by teeodor** [reply]
Trovato, bisogna installare un componente aggiuntivo, adesso però mi devo leggere bene cosa c'è da fare, anche perché mi sembra che non sia più aggiornato
https://community.home-assistant.io/t/100-templatable-lovelace-configurations/105241/74
